### ISSUE TITLE
Type inference in linear solve

### ISSUE THREAD

**Issue by gdalle** [gdalle/ImplicitDifferentiation.jl#6]
At the moment, since IterativeSolvers can return a [convergence history](https://iterativesolvers.julialinearalgebra.org/dev/getting_started/#log-keyword) depending on a keyword argument, our pushforward and pullback are type-unstable.

If this isn't fixed by #5, we can consider annotations like
```julia
linear_solver(A, b)::AbstractVector = gmres(A, b)
```
or
```julia
u::AbstractVector = linear_solver(A, b)
```

**Comment by tpapp** [comment]
Wasn't this fixed by 7072331549eb4a1e5bcb257c50f3239fcc8e5d7b?

**Comment by gdalle** [comment]
Correct, thanks!
